### DISCUSSION TITLE
"Could not transform" warnings

### DISCUSSION THREAD

**Discussion by ahoward-cf** [manusimidt/py-xbrl#92] [Q&A] [answered]
I am getting some "Could not transform" warnings and I dont really understand what that means. Are values supposed to be combined with decimals or something? I'm not sure what the intended functionality should be. Sorry for the vague question. I am new to both the package and to XBRL files in general. 

![image](https://user-images.githubusercontent.com/29404906/197992285-2a230583-8e4a-4fe6-b867-931d7688c720.png)

**Comment by manusimidt** [comment, marked as answer]
Hey, 
Normal XBRL submissions always have separate files for displaying the content to the user (pdf, html) and for programmatically reading the data (xml). In iXBRL on the other hand both is combined into one file (xHTML). Thus you get the problem that maybe you do not want your user to read the fact-value as it is. 

For example, the filer might have the sentence: "Our revenue increased to $14.3 Mrd. USD which I like". 
The corresponding xHTML would look like the following: 

```xml
<p>
   Our revenue increased to $
            <ix:nonFraction id="fact3"
                            name="example:Revenue"
                            contextRef="END_2020"
                            unitRef="usd"
                            scale="9"
                            format="ixt:numdotdecimal">
                14.3
            </ix:nonFraction>
   Mrd. USD which I like
</p>
```

`py-xbrl` then graps the value (14.3) and transforms it according to the format (ixt:numdotdecimal) and the scale to "14300000000". 

The transformation of these values is defined in so-called transformation registries. There are a lot of transformation registries and I am not able to implement them all. Thus if the transformation registry (like in your case) is not known, `py-xbrl` can not transform the fact and will just let it as it is (to give another example: the string "27th of October 2022" will not be transformed into "2022-10-27".

**Reply by manusimidt** [reply]
The transformations are defined in https://github.com/manusimidt/py-xbrl/blob/main/xbrl/transformations/__init__.py

**Comment by ahoward-cf** [comment]
Thanks for this. That does explain a lot and with that, I might be able to form my own repo and start adding some of the transformations that are more UK accounts relevant.
